internal/app/pactproxy/proxy.go

```diff
@@ -25,157 +26,146 @@ var supportedMediaTypes = map[string]func([]byte, *url.URL) (requestDocument, er
+	e.POST("/interactions/modifiers", a.interactionsModifiersHandler)
+
+	e.DELETE("/session", a.sessionHandler)
+
+	e.POST("/interactions", a.interactionsPostHandler)
+	e.DELETE("/interactions", a.interactionsDeleteHandler)
+
+	e.GET("/interactions/wait", a.interactionsWaitHandler)
+
+	e.Any("/*", a.indexHandler)
 }
 
-func (a *api) interactionsConstraintsHandler(res http.ResponseWriter, req *http.Request) {
-	constraintBytes, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		httpresponse.Errorf(res, http.StatusBadRequest, "unable to read constraint. %s", err.Error())
-		return
-	}
+func (a *api) proxyPassHandler(c echo.Context) error {
+	return a.ProxyRequest(c)
+}
+
+func (a *api) readinessHandler(c echo.Context) error {
+	return c.NoContent(http.StatusOK)
+}
 
-	constraint, err := LoadConstraint(constraintBytes)
+func (a *api) interactionsConstraintsHandler(c echo.Context) error {
+	constraint := interactionConstraint{}
+	err := c.Bind(&constraint)
 	if err != nil {
-		httpresponse.Errorf(res, http.StatusBadRequest, "unable to load constraint. %s", err.Error())
-		return
+		return c.JSON(http.StatusBadRequest, httpresponse.Errorf("unable to load constraint. %s", err.Error()))
 	}
 
 	interaction, ok := a.interactions.Load(constraint.Interaction)
 	if !ok {
-		httpresponse.Errorf(res, http.StatusBadRequest, "unable to find interaction. %s", constraint.Interaction)
-		return
+		return c.JSON(http.StatusBadRequest, httpresponse.Errorf("unable to find interaction. %s", constraint.Interaction))
 	}
 
 	log.Infof("adding constraint to interaction '%s'", interaction.Description)
 	interaction.AddConstraint(constraint)
-}
 
-func (a *api) interactionsModifiersHandler(res http.ResponseWriter, req *http.Request) {
-	modifierBytes, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		httpresponse.Errorf(res, http.StatusBadRequest, "unable to read modifier. %s", err.Error())
-		return
-	}
+	return c.NoContent(http.StatusOK)
+}
 
-	modifier, err := loadModifier(modifierBytes)
+func (a *api) interactionsModifiersHandler(c echo.Context) error {
+	modifier := &interactionModifier{}
+	err := c.Bind(modifier)
 	if err != nil {
-		httpresponse.Errorf(res, http.StatusBadRequest, "unable to load modifier. %s", err.Error())
-		return
+		return c.JSON(http.StatusBadRequest, httpresponse.Errorf("unable to load modifier. %s", err.Error()))
 	}
 
 	interaction, ok := a.interactions.Load(modifier.Interaction)
 	if !ok {
-		httpresponse.Errorf(res, http.StatusBadRequest, "unable to find interaction for modifier. %s", modifier.Interaction)
-		return
+		return c.JSON(http.StatusBadRequest, httpresponse.Errorf("unable to find interaction for modifier. %s", modifier.Interaction))
 	}
 
 	log.Infof("adding modifier to interaction '%s'", interaction.Description)
 	interaction.Modifiers.AddModifier(modifier)
+
+	return c.NoContent(http.StatusOK)
 }
 
-func (a *api) sessionHandler(res http.ResponseWriter, req *http.Request) {
-	if req.Method == http.MethodDelete {
```

Comment:
Getting rid of these `if req.Method ...` is a big win for me.  I know it's only a small thing but it seems clumsy.